src/main/business/applicationFacade.js

```diff
@@ -99,63 +97,57 @@ class ApplicationFacade{
 
     /**
      * register a user
-     * @param pid - the process trying to sign up from
      * @param userdetails - details needed to register the user with
      * @returns {Promise<User>} - returns the created user
      * @throws Error - if the process is already logged in
      *               - if the username/email is taken
      */
-    async register(pid, userdetails){
+    async register(userdetails){
```

Comment:
`register` gets `req.body` from the service, so this is not `userdetails` but much more than that. this is not what you want to send to the userController because you'll have problems. 
This is why we should not gather all the information in a single object but separate information into meaningful parameters. 
register should receive `(callingUser, registerDetails)` so we can authenticate calling user as an Admin and only then pass `registerDetails` to the `userController`

Reply:
2 things
1. I think the verification should be dont in the ```userController ```, it is a logic of the system, not the server
2. Again, I am still willing to pass only 1 until the function that using it, we wasted enoght time changing all the signature for all the 7 levels that we have in our project just to change the signature of function that pass it on.

The one that will use the variable should check that the variables are there.